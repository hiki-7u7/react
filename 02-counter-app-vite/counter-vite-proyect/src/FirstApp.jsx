import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>

      <h1> { title } </h1>
      
      <h2>{ subtitle }</h2>
      <h2>{ subtitle }</h2>
      <h2>{ subtitle }</h2>

      <h2>{ name }</h2>

    </>
  );
};

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subtitle : PropTypes.number
}

FirstApp.defaultProps = {
  name : "Sin nombre",
  subtitle: 10
}

