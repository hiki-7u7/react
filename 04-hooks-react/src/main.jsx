import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from "react-router-dom";
import './index.css';

// import { HooksApp } from './HooksApp'
// import { CounterApp } from './useState/CounterApp'
// import { CounterCustomHooks } from './useState/CounterCustomHooks'
// import { FormWithCustomHook } from './useEffect/FormWithCustomHook';
// import { MultiCustomHooks } from './03-examples/MultiCustomHooks';
// import { Counter } from './04-memo/Counter';
// import { Padre } from './07-tarea-memo/Padre';
// import './08-reducer/intro-reducer';
// import { TodoApp } from './08-reducer/TodoApp';
import { MainApp } from './09-context/MainApp';


ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    < BrowserRouter>
    
      <MainApp />

    </ BrowserRouter>

  //</React.StrictMode>,
)
