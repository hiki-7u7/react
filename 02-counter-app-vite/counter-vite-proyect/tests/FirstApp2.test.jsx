const { render , screen } = require("@testing-library/react")
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp2 />', () => { 

    const title = 'Hola soy goku';
    const subtitle = 100
    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <FirstApp title={ title } />);
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el mensaje "Hola soy Goku"', () => {

        render( <FirstApp title={ title }/>);       
        expect( screen.getByText(title) ).toBeTruthy();
 
    });

    test('debe de mostrar el mensaje en un h1', () => {

        render( <FirstApp title={ title }/>);       
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(title);
 
    });

    test('debe de contener todos los subtitulos', () => {

        render( <FirstApp title={ title } subtitle={100} />);       
        expect( screen.getAllByText(subtitle).length ).toBe(3);
 
    })

});