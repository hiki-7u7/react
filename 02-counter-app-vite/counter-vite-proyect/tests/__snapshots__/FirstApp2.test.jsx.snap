// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    Hola soy goku
     
  </h1>
  <h2>
    10
  </h2>
  <h2>
    10
  </h2>
  <h2>
    10
  </h2>
  <h2>
    Sin nombre
  </h2>
</div>
`;
